### PR TITLE
Convert dashes to underscores in cluster name

### PR DESCRIPTION
**Pull request by yenicapotediaz**
Added a fact in helm.yaml to convert dashes to underscores in cluster name to use in helm_override.
Updated schema to disallow underscores in cluster name.

Fixes https://github.com/samsung-cnct/k2/issues/513

### CODE REVIEW

File: schemas/config/v1/cluster.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "name": {
       "description": "Name of this cluster.",
       "type": "string",
+      "pattern": "^[A-Za-z0-9-]+$",
       "maxLength": 13,
```
Comment (coffeepac) [inline schemas/config/v1/cluster.json]: while you're in this file, can you take care of #624?  the maxLength is a function of how long an ELB name can be.  

new value:  32

Comment (yenicapotediaz) [inline schemas/config/v1/cluster.json]: I was going to do it in a separate pr, but with your blessing I shall do it in this one.

**Comment by davidewatson** [comment]
The first job failed due to canal errors:

> [aws] TASK [/home/jenkins/workspace/K2_PR-733-NWIPZY7GN5GDZTOUSRPUMOOULLFNEOLZKCFQQ6ZLOT64BYWH4ZUA/ansible/roles/kraken.fabric/kraken.fabric.flannel : Deploy canal daemonset] ***
[aws] fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["/opt/cnct/kubernetes/v1.7/bin/kubectl", "--kubeconfig=/home/jenkins/workspace/K2_PR-733-NWIPZY7GN5GDZTOUSRPUMOOULLFNEOLZKCFQQ6ZLOT64BYWH4ZUA/cluster/aws//pr-733-1/admin.kubeconfig", "apply", "-f", "/home/jenkins/workspace/K2_PR-733-NWIPZY7GN5GDZTOUSRPUMOOULLFNEOLZKCFQQ6ZLOT64BYWH4ZUA/cluster/aws//pr-733-1/fabric/canal.yaml", "--namespace=kube-networking"], "delta": "0:01:00.265510", "end": "2017-08-09 23:51:58.419353", "failed": true, "rc": 1, "start": "2017-08-09 23:50:58.153843", "stderr": "error: error when retrieving current configuration of:\n&{0xc420eb0180 0xc42010d810 kube-networking canal /home/jenkins/workspace/K2_PR-733-NWIPZY7GN5GDZTOUSRPUMOOULLFNEOLZKCFQQ6ZLOT64BYWH4ZUA/cluster/aws//pr-733-1/fabric/canal.yaml 0xc420e17b70 0xc420e17b70  false}\nfrom server for: \"/home/jenkins/workspace/K2_PR-733-NWIPZY7GN5GDZTOUSRPUMOOULLFNEOLZKCFQQ6ZLOT64BYWH4ZUA/cluster/aws//pr-733-1/fabric/canal.yaml\": Get https://pr-733-1-master-elb-1347830143.us-east-2.elb.amazonaws.com/api/v1/namespaces/kube-networking/serviceaccounts/canal: unexpected EOF", "stderr_lines": ["error: error when retrieving current configuration of:", "&{0xc420eb0180 0xc42010d810 kube-networking canal /home/jenkins/workspace/K2_PR-733-NWIPZY7GN5GDZTOUSRPUMOOULLFNEOLZKCFQQ6ZLOT64BYWH4ZUA/cluster/aws//pr-733-1/fabric/canal.yaml 0xc420e17b70 0xc420e17b70  false}", "from server for: \"/home/jenkins/workspace/K2_PR-733-NWIPZY7GN5GDZTOUSRPUMOOULLFNEOLZKCFQQ6ZLOT64BYWH4ZUA/cluster/aws//pr-733-1/fabric/canal.yaml\": Get https://pr-733-1-master-elb-1347830143.us-east-2.elb.amazonaws.com/api/v1/namespaces/kube-networking/serviceaccounts/canal: unexpected EOF"], "stdout": "daemonset \"canal\" created", "stdout_lines": ["daemonset \"canal\" created"]}

And the 2nd job failed due to an RBAC error. I know @mikeln was working on an RBAC problem today (#728), but I am unsure if it is related.

> [aws] fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["/opt/cnct/kubernetes/v1.7/bin/kubectl", "--kubeconfig=/home/jenkins/workspace/K2_PR-733-NWIPZY7GN5GDZTOUSRPUMOOULLFNEOLZKCFQQ6ZLOT64BYWH4ZUA/cluster/aws//pr-733-2/admin.kubeconfig", "apply", "-f", "/home/jenkins/workspace/K2_PR-733-NWIPZY7GN5GDZTOUSRPUMOOULLFNEOLZKCFQQ6ZLOT64BYWH4ZUA/cluster/aws//pr-733-2/auth/rbac-default-policy.yaml"], "delta": "0:00:00.465349", "end": "2017-08-09 23:54:11.964739", "failed": true, "rc": 1, "start": "2017-08-09 23:54:11.499390", "stderr": "unable to recognize \"/home/jenkins/workspace/K2_PR-733-NWIPZY7GN5GDZTOUSRPUMOOULLFNEOLZKCFQQ6ZLOT64BYWH4ZUA/cluster/aws//pr-733-2/auth/rbac-default-policy.yaml\": no matches for rbac.authorization.k8s.io/, Kind=ClusterRole\nunable to recognize \"/home/jenkins/workspace/K2_PR-733-NWIPZY7GN5GDZTOUSRPUMOOULLFNEOLZKCFQQ6ZLOT64BYWH4ZUA/cluster/aws//pr-733-2/auth/rbac-default-policy.yaml\": no matches for

**Comment by mikeln** [comment]
That looks like the other error we have seen in 1.7.x when setting RBAC via a yaml file #703 ,
however in this case it appears to not just be the rbac.yaml that has the issue, but the regular configuration `canal.yaml`.     I'm starting to wonder if kubectl for 1.7.x has some issue?

**Comment by coffeepac** [comment]
I'm going to push this through even with the not-passed CI.  This is going to be breaking k2-tools CI for a bit without it.
